src/commands/slow-mode.ts

```diff
@@ -0,0 +1,599 @@
+		where: {
+			channel_id: channel.id
+		},
+		include: 'stages',
+		defaults: {
+			channel_id: channel.id,
+			enabled: false
+		}
+	});
+
+	const threshold = interaction.options.getInteger('threshold', true);
+	const limit = interaction.options.getInteger('limit', true);
+
+	let oldLimit: number | null = null;
+	let stage = slowMode.stages?.find(stage => stage.threshold === threshold);
+	if (stage) {
+		oldLimit = stage.limit;
+		stage.limit = interaction.options.getInteger('limit', true);
+		await stage.save();
+	} else {
+		stage = await SlowModeStage.create({
+			threshold,
+			limit
+		});
+		await slowMode.addStage(stage);
+		await slowMode.reload({ include: 'stages' });
+	}
+
+	const auditLogEmbed = new EmbedBuilder()
+		.setColor(0xC0C0C0)
+		.setTitle('Event Type: _Auto Slow-Mode Threshold Set_')
+		.setDescription('――――――――――――――――――――――――――――――――――')
+		.setFields(
+			{
+				name: 'User',
+				value: `<@${interaction.user.id}>`
+			},
+			{
+				name: 'Channel',
+				value: `<#${channel.id}>`
+			},
+			{
+				name: 'New Threshold',
+				value: `Limit of 1 message every ${limit} seconds above ${threshold} messages per minute`
+			}
+		).setFooter({
+			text: 'Pretendo Network',
+			iconURL: interaction.guild!.iconURL()!
+		});
+
+	if (oldLimit) {
+		auditLogEmbed.addFields([
+			{
+				name: 'Old Threshold',
+				value: `Limit of 1 message every ${oldLimit} seconds above ${threshold} messages per minute`
+			}
+		]);
+	}
+
+	if (slowMode.stages && slowMode.stages.length > 0) {
+		const stagesMessage = slowMode.stages.sort((a, b) => a.threshold - b.threshold).map(stage => {
+			return `1 message every ${stage.limit}s above ${stage.threshold} messages per minute`;
+		}).join('\n');
+	
+		auditLogEmbed.addFields([
+			{
+				name: 'Configured stages',
+				value: stagesMessage
+			}
+		]);
+	}
+		
+	await sendEventLogMessage(interaction.guild!, channel.id, auditLogEmbed);
+
+	await interaction.followUp({ content: `Set a limit of 1 message every ${limit} seconds above ${threshold} messages per minute on <#${channel.id}>` });
+}
+
+async function unsetStageHandler(interaction: ChatInputCommandInteraction): Promise<void> {
+	await interaction.deferReply({ ephemeral: true });
+
+	const channel = interaction.options.getChannel('channel') ?? interaction.channel;
+	if (!channel) {
+		throw new Error('No channel given');
+	}
+
+	if (channel.type !== ChannelType.GuildText) {
+		throw new Error('Slow mode only applies to text channels');
+	}
+
+	const slowMode = await SlowMode.findOne({
+		where: {
+			channel_id: channel.id
+		},
+		include: 'stages'
+	});
+
+	if (!slowMode) {
+		throw new Error(`No slow mode set for <#${channel.id}>`);
+	}
+
+	const threshold = interaction.options.getInteger('threshold', true);
+
+	const stage = slowMode.stages?.find(stage => stage.threshold === threshold);
+	if (!stage) {
+		throw new Error(`No stage set at ${threshold} messages per minute`);
+	}
+
+	const oldLimit = stage.limit;
+	await slowMode.destroy();
```

Comment:
This destroys the slowmode configuration, not the stage.

Reply:
I was definitely too tired doing this 😂 at least it would probably also destroy the stage!